src/string.cr

```diff
@@ -611,6 +641,133 @@ class String
     ToU64Info.new value, negative, invalid
   end
 
+  # :nodoc:
+  record ToU128Info,
+    value : UInt128,
+    negative : Bool,
+    invalid : Bool
+
+  private macro gen_to_128(method, max_positive = nil, max_negative = nil)
+    info = to_u128_info(base, whitespace, underscore, prefix, strict)
+    return yield if info.invalid
+
+    if info.negative
+      {% if max_negative %}
+        return yield if info.value > {{max_negative}}
+        -info.value.to_{{method}}
```

Comment:
The same issues expressed in #7172 apply here.

Reply:
good call I will update this.